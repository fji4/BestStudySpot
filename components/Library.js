
import React, { Component } from 'react';
import axios from 'axios';
import {
    Platform,
    StyleSheet,
    NavigatorIOS,
    View,
    Image,
    ScrollView
} from 'react-native';
import  {
    Container,
    Header,
    Content,
    Footer,
    FooterTab,
    Button,
    Text,
    Badge,
    Thumbnail,
    List,
    ListItem,
    Left,
    Right,
    Separator,
    Form,
    Item,
    Label,
    Input,
    Card,
    CardItem,
    Body
} from 'native-base';

import Icon from 'react-native-vector-icons/MaterialIcons';
import {Actions} from 'react-native-router-flux';


class LibraryListItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            library: {}
        };

        this.get_library();
    }



    get_library() {
        axios.get(this.props.library['unit_json'])
            .then(function (response) {
                console.log(response.data);
                this.setState({
                    library:response.data[0]
                });
        }.bind(this));
    }



    render() {
        return(
            <Card style={{flex: 0}} >

                <CardItem>
                    <Left>
                        <Body>
                        <Text onPress={() => Actions.webview({html:this.state.library['weblinks']})}>
                            {this.props.library['library_name']}
                            </Text>
                        <Text note>{this.state.library['street_address']}</Text>
                        </Body>
                    </Left>
                </CardItem>
                <CardItem>
                    <Body>
                    <Image source={{uri: this.state.library['parking_image']}} style={{height: 200, width: 200, flex: 1}}/>
                    <Text>{this.state.library['description']}</Text>
                    </Body>
                </CardItem>
            </Card>
        )
    }
}

/**
 * Render the whole list of the followers
 * @param props
 * @returns {XML}
 * @constructor
 */
const LibraryList = props => {
    const LibraryItems = props.libraries.map(library => {
        return (
            <LibraryListItem
                key = {library['unit_id']}
                library={library}
            />
        );
    });

    return (
        <ScrollView>
            {LibraryItems}
        </ScrollView>
    );
};




export default class Library extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            libraries:[]
        };

        this.getAuthInformation();
    }

    /**
     * Call library api to get all the user information and store them to a dic.
     */

    getAuthInformation() {
        axios.get("https://quest.library.illinois.edu/LibDirectory/Api/UnitsWithCalendars")
            .then(function(response){
                this.setState({
                    libraries:response.data
                },function () {

                });
            }.bind(this));
    }

    render() {
        return(
            <Container>
                <LibraryList libraries={this.state.libraries}/>
                <Footer>
                    <FooterTab>
                        <Button vertical active>
                            <Icon name="local-library" size={30} />
                            <Text style={{fontSize: 10}}>Library</Text>
                        </Button>
                        <Button vertical onPress={() => Actions.replace("recommend")}>
                            <Icon name="comment" size={30}/>
                            <Text style={{fontSize: 10}}>Recommend</Text>
                        </Button>
                        <Button vertical onPress={() => Actions.comment()}>
                            <Icon active name="add-circle-outline" size={30}/>
                        </Button>
                        <Button vertical onPress={() => Actions.replace("user")}>
                            <Icon name="person" size={30}/>
                            <Text style={{fontSize: 10}}>Me</Text>
                        </Button>
                    </FooterTab>
                </Footer>
            </Container>
    )}
}