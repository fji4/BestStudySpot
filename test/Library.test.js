/**
 * Created by aliceji on 12/5/17.
 */
import React from 'react';
import Library from '../components/Library';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(
        <Library />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the first library', () => {
    const ListView = require('ListView');
    const Text = require('Text');
    const dataSource = new ListView.DataSource({
        rowHasChanged: (r1, r2) => r1 !== r2,
    }).cloneWithRows(['Grainger Engineering Library Information Center', '1301 West Springfield Avenue']);
    const tree = renderer
        .create(
            <ListView
                dataSource={dataSource}
                renderRow={rowData => <Text>{rowData}</Text>}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders the second library', () => {
    const ListView = require('ListView');
    const Text = require('Text');
    const dataSource = new ListView.DataSource({
        rowHasChanged: (r1, r2) => r1 !== r2,
    }).cloneWithRows(['Mathematical Library', '1409 West Green Street']);
    const tree = renderer
        .create(
            <ListView
                dataSource={dataSource}
                renderRow={rowData => <Text>{rowData}</Text>}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});