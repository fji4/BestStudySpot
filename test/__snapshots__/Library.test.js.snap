// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderColor": "#cbcbcb",
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "left": 0,
        "paddingBottom": 0,
        "right": 0,
      }
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": undefined,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
      virtual={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#cde1f9",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#007aff",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          Library
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          Recommend
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 0,
            "flex": 1,
            "flexDirection": "column",
            "height": null,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 10,
              },
            ]
          }
          uppercase={false}
          virtual={undefined}
        >
          Me
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders the first library 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Grainger Engineering Library Information Center
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      1301 West Springfield Avenue
    </Text>
  </View>
</RCTScrollView>
`;

exports[`renders the second library 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Mathematical Library
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      1409 West Green Street
    </Text>
  </View>
</RCTScrollView>
`;
